users.js

```diff
@@ -95,6 +96,86 @@ var lockedUsers = Object.create(null);
 var lockedRanges = Users.lockedRanges = Object.create(null);
 var rangelockedUsers = Object.create(null);
 
+function importLocksBans() {
+	fs.readFile('config/locksbans.csv', function (err, data) {
+		if (err) return;
+		var now = Date.now();
+		data = ('' + data).split("\n");
+		for (var i = 0; i < data.length; i++) {
+			if (!data[i]) continue;
+			var row = data[i].split(",");
+			var expiration = parseInt(row[2]);
+			if (now > expiration) continue;
+			var entry = [row[1], expiration];
+			if (row[0] === 'banip') {
+				bannedIps[row[3]] = entry;
+			} else if (row[0] === 'ban') {
+				bannedUsers[row[3]] = entry;
+			} else if (row[0] === 'lockip') {
+				lockedIps[row[3]] = entry;
+			} else if (row[0] === 'lock') {
+				lockedUsers[row[3]] = entry;
+			}
+		}
+	});
+
+	fs.readFile('config/ipbans.txt', function (err, data) {
+		if (err) return;
+		data = ('' + data).split("\n");
+		var rangebans = [];
+		for (var i = 0; i < data.length; i++) {
+			data[i] = data[i].split('#')[0].trim();
+			if (!data[i]) continue;
+			if (data[i].indexOf('/') >= 0) {
+				rangebans.push(data[i]);
+			} else if (!Users.bannedIps[data[i]]) {
+				Users.bannedIps[data[i]] = '#ipban';
+			}
+		}
+		Users.checkRangeBanned = Cidr.checker(rangebans);
+	});
+}
+function exportLocksBans() {
+	var buffer = '';
+	var now = Date.now();
+	for (var ip in bannedIps) {
+		if (!Array.isArray(bannedIps[ip])) continue;
+		if (now > bannedIps[ip][1]) {
+			delete bannedIps[ip];
+			continue;
+		}
+		buffer += "banip," + bannedIps[ip][0] + "," + bannedIps[ip][1] + "," + ip + "\n";
+	}
+	for (var user in bannedUsers) {
+		if (now > bannedUsers[user][1]) {
+			delete bannedUsers[user];
+			continue;
+		}
+		buffer += "ban," + bannedUsers[user][0] + "," + bannedUsers[user][1] + "," + user + "\n";
+	}
+	for (var ip in lockedIps) {
+		if (!Array.isArray(lockedIps[ip])) continue;
+		if (now > lockedIps[ip][1]) {
+			delete lockedIps[ip];
+			continue;
+		}
+		buffer += "lockip," + lockedIps[ip][0] + "," + lockedIps[ip][1] + "," + ip + "\n";
+	}
+	for (var user in lockedUsers) {
+		if (now > lockedUsers[user][1]) {
+			delete lockedUsers[user];
+			continue;
+		}
+		buffer += "lock," + lockedUsers[user][0] + "," + lockedUsers[user][1] + "," + user + "\n";
+	}
+	fs.writeFileSync('config/locksbans.csv', buffer);
+}
+Users.importLocksBans = importLocksBans;
+Users.exportLocksBans = exportLocksBans;
+Users.backupLocksBansTimer = setInterval(function () {
+	exportLocksBans();
+}, 12 * 60 * 60 * 1000);
```

Comment:
You know you can just do

    setInterval(exportLocksBans, 12 * 60 * 60 * 1000);

...right?